packages/augur-ui/src/services/ipfs/IPFS.worker.ts

```diff
@@ -0,0 +1,185 @@
+import IPFS from 'ipfs';
+import Orbit from 'orbit_';
+
+declare var self: Worker;
+
+const ORBIT_EVENTS = ['connected', 'disconnected', 'joined', 'left', 'peers'];
+const CHANNEL_FEED_EVENTS = [
+  'write',
+  'load.progress',
+  'replicate.progress',
+  'ready',
+  'replicated',
+];
+
+// this to be as typesafe as possible with self and addEventListener + postMessage
+const ctx = self;
+let ipfs: IPFS = null;
+let orbit: Orbit = null;
+let currentChannelName = '';
+let currentChannel = null;
+let ethereumNetwork = '';
+
+ctx.addEventListener(
+  'message',
+  async (message: { data: { method: string; params: any[] } }) => {
+    const messageData = message.data;
+
+    try {
+      console.log('IPFS.worker', messageData);
+
+      switch (messageData.method) {
+        case 'IPFS:start': {
+          ethereumNetwork = messageData.params[0] || 'dev';
+          const senderAccount = messageData.params[1] || 'augur-ipfs';
+
+          const ipfsOptions = {
+            repo: `augur-orbit-chat-ipfs-${ethereumNetwork}`,
+            EXPERIMENTAL: {
+              pubsub: true,
+            },
+            preload: { enabled: false },
+            config: {
+              Addresses: {
+                Swarm: [
+                  '/dns4/ws-star.discovery.libp2p.io/tcp/443/wss/p2p-websocket-star',
+                ],
+              },
+            },
+          };
+
+          const orbitOptions = {
+            directory: `augur-orbit-chat-orbitdb-${ethereumNetwork}`,
+            id: senderAccount,
+          };
+
+          ipfs = await IPFS.create(ipfsOptions);
+          orbit = await Orbit.create(ipfs, orbitOptions);
+
+          // subscribeOrbitEvents.bind(this)();
+
+          orbit.events.emit('connected', (orbit as any).user);
+
+          trackNumberOfPeers();
+
+          ctx.postMessage({ method: 'IPFS:started' });
+
+          break;
+        }
+        case 'IPFS:sendMessage': {
+          const [data, channel] = messageData.params;
+
+          if (!data || !channel) break;
+
+          if (currentChannelName !== channel || currentChannel == null) {
+            if (currentChannelName) {
+              await orbit.leave(currentChannelName);
+            }
+
+            currentChannelName = `${channel}-network-${ethereumNetwork}`;
+
+            currentChannel = await orbit.join(currentChannelName);
+
+            // subscribeOrbitChannelEvents.bind(this)(
+            //   currentChannel,
+            //   currentChannelName
+            // );
```

Comment:
If this is going to be used in the future, I suggest adding it later instead of commenting it